dask/array/fft.py

```diff
@@ -28,31 +30,37 @@ def _fft_out_chunks(a, n, axis):
     return chunks
 
 
-def _rfft_out_chunks(a, n, axis):
+def _rfft_out_chunks(a, n, axis, rtype):
     if n is None:
         n = a.chunks[axis][0]
     chunks = list(a.chunks)
-    chunks[axis] = (n // 2 + 1,)
+    if issubclass(np.dtype(rtype).type, numbers.Real):
+        chunks[axis] = (n,)
+    else:
+        chunks[axis] = (n // 2 + 1,)
```

Comment:
Is it reasonable for someone to come in with a new fft module and it return a complex dtype but not follow scipy's shape scheme?  My question is "are we using dtype to infer which module this is coming from?"  If so then we should probably just condition on the module itself rather than back out which module is being used based on what is happening to the dtype.

Taking a step back, is there a way to better handle the shape business here?  This all seems somewhat convoluted.